go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -538,9 +538,22 @@ func (c *Controller) onANPPodUpdate(oldObj, newObj interface{}) {
 
 // onANPPodDelete queues the pod for processing.
 func (c *Controller) onANPPodDelete(obj interface{}) {
-	key, err := cache.MetaNamespaceKeyFunc(obj)
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
```

Comment:
so this happens when somehow we missed the deletion event due to disconnection from apiserver perhaps for whatever reason and the object gets placed into the cache and its possible this object is stale but still there is a chance we could trigger deletion based on that huh?